upgrade/v1beta1/upgrade.go

```diff
@@ -94,6 +94,9 @@ func fixupVolumes(namespace string, lhClient *lhclientset.Clientset) error {
 	for _, obj := range volumes.Items {
 		existing := obj.DeepCopy()
 
+		if obj.Spec.ReplicaAutoBalance == "" {
```

Comment:
I'd recommend move this to v123 to v124 upgrade path and back port to 1.2.4 branch as well.

Reply:
Sounds good but it means need to use v1beta1 in the 123 to 124 upgrade path for this update? and it’s nice to have that in 1.2.4.

Reply:
Yes, and it uses v1beta1 API to operate now.

Reply:
We still need to put the ReplicaAutoBalance in `upgrade/v1beta1/upgrade.go`, because `doCRUpgrade` is executed after `doAPIVersionUpgrade`. `doCRUpgrade` uses v1beta2 now, so the `volume` updateStatus failure occurs in v1.1.0 to v1.2.0 upgrade.

```
time="2022-01-16T08:19:37Z" level=info msg="upgrade from v1.1.0 to v1.2.0: Updating pvc-dd63e559-6809-4c40-881f-3b98ce382d02 volume labels to map[longhornvolume:pvc-dd63e559-6809-4c40-881f-3b98ce382d02 recurring-job.longhorn.io/backup-20-b5f4656e-77074ba4:enabled]" action=convert-volume-recurring-jobs-to-labels namespace=longhorn-system translate=volume-recurring-jobs
```

Another way is moving ReplicaAutoBalance to the conversion webhook. WDYT?

Reply:
This is the way to do, but I would prefer making the conversion webhook focusing on the things which can't handled in the migration like non-backward compatibility. 

You should just create a `v123to124/upgrade.go`, then use `lhClient.LonghornV1beta1()` to do the migration to update ReplicaAutoBalance to the default value?

Reply:
> You should just create a v123to124/upgrade.go, then use lhClient.LonghornV1beta1() to do the migration to update ReplicaAutoBalance to the default value?

No... The `volume` read from `lhClient.LonghornV1beta1` does not have the `Spec.ReplicaAutoBalance` field, so the migration is not achievable by this way.

Reply:
Sorry. I mixed up the v1beta1 structs in v1.1.3 and master.
The current hurdle is the volume update fails on v110tov120 unless we replace all v1beta2 APIs with v1beta2 APIs in upgrade logics. The upgrade logics use datastore APIs only accepting v1beta2 objects.

So any migration impacted by validation, we now just put into fixxxxx only for the current implementation. We don't need to have backport now, because at the end, when users upgrade to 1.3.0, it will still be migrated

Reply:
Thanks @derekbit , all fine now.